main.rb

```diff
@@ -1 +1,109 @@
+    login = ""
+    password = ""
+    username = @browser.text_field(:class=>"username")
+    username.exists?
+    username.set login
+    username.value
+    pass = @browser.text_field(id:"password")
+    pass.exists?
+    pass.set password
+    pass.value
+    @browser.button(:class=>"wb-button").click
+  end
+
+
+  public def accountInfoExtract
+    @browser.div(:class=>"block__main-menu").wait_until(&:present?)
+    @browser.span(:text=>"Carduri și conturi").click
+    @browser.div(:class=>"contracts-section").wait_until(&:present?)
+    @accounts_array=Array.new
+
+    doc = Nokogiri::HTML.parse(@browser.html)
+    doc.css("div.main-info").each do |acc|
+      name = acc.css("a.name").text
+      currency = acc.css("div.icon.icon-account").text
+      amount = acc.css("span.amount")[0].text
+      account = Accounts.new(name,currency,amount)
+      @accounts_array.push(account)
+    end
+    @browser.divs(:class=>"main-info").map do |acc|
+      name = acc.link(:class=>"name").title
+      acc.link(:class=>"name").click
+      @browser.link(:href=>"#contract-history").wait_until(&:present?).click
+      sleep 3
+      setPeriod
+      @transaction_array= Array.new
+      account = @accounts_array.detect{|a| a.name.casecmp?(name)}
+      transactionsInfoByAccountExtract(name,account)
+      account.transactions=@transaction_array
+      @browser.span(:text=>"Carduri și conturi").click
+      @browser.div(:class=>"contracts-section").wait_until(&:present?)
+    end
+  end
+
+  def setPeriod
+    @browser.link(:href=>"#contract-history").wait_until(&:present?).click
+    @browser.div(:class=>"filters").wait_until(&:present?)
+    time = Date.today << 2
+    @browser.input(:name=>"from").click
+    i = 1
+    @browser.div(:class=>"ui-datepicker-title ").browser.divs(:class=>"arrow").map do |arrow|
+      if i == 3 then
+        break
+      elsif i == 2 then
+        arrow.wait_until(&:present?).click
+        @browser.li(:text=>(time.year).to_s).click
+      end
+      i += 1
+    end
+    @browser.div(:class=>"ui-datepicker-title ").div(:class=>"arrow").wait_until(&:present?).click
+    @browser.li("data-option-array-index"=>(time.month-1).to_s).click
+    @browser.link(:text=>time.day.to_s).click
+    sleep 2
+  end
+
+  def transactionsInfoByAccountExtract(acc,account)
+    if @browser.link(:class=>"operation-details").exists?
+      @browser.spans(:class=>"history-item-description").map do |trans|
+        data, description, amount, currency=transactionsInfoExtract(trans)
+        @transaction_array.push(Transactions.new(data, description, amount, currency, acc))
+      end
+    end
+  end
+
+  def transactionsInfoExtract(trans)
+     trans.link(:class=>"operation-details").wait_until(&:present?).click
+     @browser.div(:class=>"details-section").wait_until(&:present?)
+     details = Nokogiri::HTML.parse(@browser.html)
+     data = details.css("div.details-section")[0].css("div.value").text
```

Comment:
You can use `at_css` to take first element from CSS array. 
```
data = details.at_css("div.details-section").at_css("div.value").text
```

Reply:
You can use at_css to take first element from CSS array.

data = details.at_css("div.details-section div.value").text